db/pebble/batch.go

```diff
@@ -110,10 +110,19 @@ func (b *batch) Write() error {
 	if b.db.closed {
 		return pebble.ErrClosed
 	}
-	return b.batch.Commit(pebble.Sync)
+
+	if err := b.batch.Commit(pebble.Sync); err != nil {
+		return err
+	}
+
+	return b.batch.Close()
 }
```

Comment:
Could you please add a new a test asserting the new behaviour of Write

Reply:
I can add more assertion to check that a subsequent `Close()` call will throw an error, but there isn't anything else to check, because the close status is not exposed by pebble.

Reply:
Actually it's unexpected behaviour to call Close twice, because it the database doesn't need to retain anything, the first `Close()` call immediately recycle the batch to be used in future `NewBatch` call, so we can't really test it here.